pkg/api/types.go

```diff
@@ -1414,6 +1414,38 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	unversioned.TypeMeta `json:",inline"`
+	ObjectMeta           `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// The minimum number of pods that must be available simultaneously.  Exactly
+	// one of these must be specified.
+
+	// This means an absolute number of pods.
+	MinAvailable *int `json:"minAvailable,omitempty"`
+
+	// This means a percentage of the number of pods.  Must be a value in the
+	// range {0,100}.  At evaluation time, the target will be rounded up to the
+	// next nearest whole number.
+	MinAvailablePercentage *int `json:"minAvailablePercentage,omitempty"`
+}
+
+// DisruptionBudgetStatus represents information about the status of a DisruptionBudget. Status may trail the actual
+// state of a system.
+type DisruptionBudgetStatus struct {
```

Comment:
No objection.  Is this basically so that users have a little more context to understand why the bit is T or F?